index.js

```diff
@@ -98,6 +112,37 @@ function Collection(manager, name){
 
 Collection.prototype.__proto__ = MonkCollection.prototype;
 
+Collection.prototype.ensureEventEmitterExists = function() {
+  if ( ! this._eventEmitter ) {
+    this._eventEmitter = new EventEmitter();
+  }
+};
+
+Collection.prototype.on = function() {
+  this.ensureEventEmitterExists();
+  return this._eventEmitter.on( ...arguments );
```

Comment:
If the above comment is true and we choose to make emiter a lazy getter, this and the following can be replaced with 
```js
  this.getEventEmitter().on( ...arguments );
```